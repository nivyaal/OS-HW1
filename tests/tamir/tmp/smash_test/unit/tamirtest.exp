smash> smash> smash: got ctrl-Z
smash: process \d+ was stopped
smash> \[1\] timeout 110 sleep 20 & : \d+ \d+ secs
\[2\] sleep 150 : \d+ \d+ secs \(stopped\)
smash> smash: got ctrl-C
smash: process \d+ was killed
smash> \[1\] timeout 110 sleep 20 & : \d+ \d+ secs
\[2\] sleep 150 : \d+ \d+ secs \(stopped\)
smash> smash> klfjfdsfkjdsfjkkjfkjlewfkjjkl erkjlf
f;oksdfljdgjkdfasd
flkdslfvker
f lofkopewrf
 f ofdopd 
 
 fvofpokrkp
 
 kofdkopkfoew
 
 olfgkrokl;kl;k;lkl;fd
 kkkkkk490r943ioijojifgjfiofgiojrfiowiofo
 likfdhgjihofioehr
 pofdgpoerpojfjp43
 ;lkkrkjfjovojijiorlkjknjnfklgerkj
 098vuu3jfjifiioriohufo
 tamirtest lkjfekjlfjk
 skljdfjkldsfjklsd
 tamirtest fkljsdkfjksd
 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3diojsffffffffffffffffffffffffffffffffffffffffffr83489ru834
 pofidofii349r34utfujre
 flodsoksopifiowesmash> smash>  tamirtest lkjfekjlfjk
 tamirtest fkljsdkfjksd
smash> smash> tamirtest 1234                                                                               
tamirtest 1234
tamirtest 1234
tamirtest 5678                                                                               
tamirtest 5678                                                                               
tamirtest 5678                                                                               
tamirtest 1234
tamirtest 5678                                                                               
tamirtest 5678                                                                               
tamirtest 1234
smash> tamirtest 1234                                                                               
tamirtest 1234
tamirtest 1234
tamirtest 5678                                                                               
tamirtest 5678                                                                               
tamirtest 5678                                                                               
tamirtest 1234
tamirtest 5678                                                                               
tamirtest 5678                                                                               
tamirtest 1234smash> smash> smash> smash> \[1\] timeout 110 sleep 20 & : \d+ \d+ secs
\[2\] sleep 150 : \d+ \d+ secs \(stopped\)
\[3\] timeout 15 sleep 5& : \d+ \d+ secs
\[4\] timeout 10 sleep 16& : \d+ \d+ secs
\[5\] timeout 30 sleep 7 & : \d+ \d+ secs
smash> smash: got an alarm
smash: got an alarm
smash: timeout 10 sleep 16& timed out!
smash: got an alarm
smash: got ctrl-C
smash: got ctrl-Z
smash pid is \d+
smash> smash pid is \d+
smash> smash: got ctrl-Z
\[2\] sleep 150 : \d+ \d+ secs \(stopped\)
smash> smash: got an alarm
sleep 150 : \d+
smash: got ctrl-Z
smash: process \d+ was stopped
smash> \[2\] sleep 150 : \d+ \d+ secs \(stopped\)
smash> smash> smash> sleep 15& : \d+
smash: got ctrl-Z
smash: process \d+ was stopped
smash> \[2\] sleep 150 : \d+ \d+ secs \(stopped\)
\[3\] sleep 15& : \d+ \d+ secs \(stopped\)
\[4\] sleep 100& : \d+ \d+ secs
smash> sleep 15& : \d+
smash> \[2\] sleep 150 : \d+ \d+ secs \(stopped\)
\[3\] sleep 15& : \d+ \d+ secs
\[4\] sleep 100& : \d+ \d+ secs
smash> smash> signal number 9 was sent to pid \d+
smash> \[2\] sleep 150 : \d+ \d+ secs \(stopped\)
\[5\] timeout 20 sleep 150& : \d+ \d+ secs
smash> smash error: kill: job-id 3 does not exist
smash> smash> smash error: kill: job-id 3 does not exist
smash> smash: got an alarm
smash: timeout 20 sleep 150& timed out!
\[2\] sleep 150 : \d+ \d+ secs \(stopped\)
\[6\] sleep 200& : \d+ \d+ secs
smash> smash> smash> smash> \[2\] sleep 150 : \d+ \d+ secs \(stopped\)
\[6\] sleep 200& : \d+ \d+ secs
\[7\] sleep 100& : \d+ \d+ secs
\[8\] sleep 200& : \d+ \d+ secs
\[9\] sleep 500 & : \d+ \d+ secs
smash> signal number 19 was sent to pid \d+
smash> smash> signal number 18 was sent to pid \d+
smash> \[2\] sleep 150 : \d+ \d+ secs \(stopped\)
\[6\] sleep 200& : \d+ \d+ secs
\[7\] sleep 100& : \d+ \d+ secs
\[8\] sleep 200& : \d+ \d+ secs
\[9\] sleep 500 & : \d+ \d+ secs
\[10\] sleep 200& : \d+ \d+ secs
smash> sleep 200& : \d+
smash: got ctrl-Z
smash: process \d+ was stopped
smash> sleep 100& : \d+
smash: got an alarm
smash: got ctrl-Z
smash: process \d+ was stopped
smash> sleep 200& : \d+
smash: got ctrl-C
smash: process \d+ was killed
smash> \[2\] sleep 150 : \d+ \d+ secs \(stopped\)
\[6\] sleep 200& : \d+ \d+ secs
\[7\] sleep 100& : \d+ \d+ secs \(stopped\)
\[9\] sleep 500 & : \d+ \d+ secs
\[10\] sleep 200& : \d+ \d+ secs \(stopped\)
smash> smash> /tmp/smash_test
smash> smash error: cd: OLDPWD not set
smash> smash error: cd: OLDPWD not set
smash> /tmp/smash_test
smash> smash> smash: got an alarm
smash: timeout 10 sleep 150& timed out!
did I pass the test?
smash> smash> did I pass the test?
I hope so
smash> Bye Bye
smash> smash: sending SIGKILL signal to 5 jobs:
\d+: sleep 150
\d+: sleep 200&
\d+: sleep 100&
\d+: sleep 500 &
\d+: sleep 200&
