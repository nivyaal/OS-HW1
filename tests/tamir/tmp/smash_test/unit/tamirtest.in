timeout 110 sleep 20 &
sleep 150
!time.sleep(6)
CtrlZ
jobs
timeout 15 sleep 150
!time.sleep(6)
CtrlC
jobs
cat test_in > tamir_out
cat tamir_out
showpid >> tamir_out
cat tamir_out | grep tamirtest
cat test_in2 > tamir_out
cat tamir_out | grep tamirtest
cat tamir_out
timeout 15 sleep 5&
timeout 10 sleep 16&
timeout 30 sleep 7 &
jobs
!time.sleep(15)
CtrlC
CtrlZ
showpid &
showpid&
CtrlZ
jobs
!time.sleep(15)
fg 2
!time.sleep(5)
CtrlZ
jobs
sleep 15&
sleep 100&
fg 3
CtrlZ
!time.sleep(5)
jobs
bg 3
!time.sleep(5)
jobs
timeout 20 sleep 150&
!time.sleep(5)
kill -9 4
!time.sleep(5)
jobs
kill -19 3
!time.sleep(5)
sleep 200&
kill -18 3
!time.sleep(7)
jobs
sleep 100&
sleep 200&
sleep 500 &
jobs
kill -19 8
!time.sleep(5)
sleep 200&
kill -18 8
!time.sleep(7)
jobs
fg 10
!time.sleep(7)
CtrlZ
fg 7
!time.sleep(7)
CtrlZ
fg 8
!time.sleep(7)
CtrlC
jobs
timeout 10 sleep 150&
pwd
cd -
cd -
pwd
!time.sleep(5)
echo "did I pass the test?" > test_res
!time.sleep(5)
cat test_res |& tail -c 12
echo "I hope so" >> test_res
cat test_res
echo "Bye Bye"
!time.sleep(10)
quit kill
