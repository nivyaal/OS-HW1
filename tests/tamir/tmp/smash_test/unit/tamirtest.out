smash> smash> smash: got ctrl-Z
smash: process 7293 was stopped
smash> [1] timeout 110 sleep 20 & : 7292 7 secs
[2] sleep 150 : 7293 0 secs (stopped)
smash> smash: got ctrl-C
smash: process 7297 was killed
smash> [1] timeout 110 sleep 20 & : 7292 13 secs
[2] sleep 150 : 7293 6 secs (stopped)
smash> smash> klfjfdsfkjdsfjkkjfkjlewfkjjkl erkjlf
f;oksdfljdgjkdfasd
flkdslfvker
f lofkopewrf
 f ofdopd 
 
 fvofpokrkp
 
 kofdkopkfoew
 
 olfgkrokl;kl;k;lkl;fd
 kkkkkk490r943ioijojifgjfiofgiojrfiowiofo
 likfdhgjihofioehr
 pofdgpoerpojfjp43
 ;lkkrkjfjovojijiorlkjknjnfklgerkj
 098vuu3jfjifiioriohufo
 tamirtest lkjfekjlfjk
 skljdfjkldsfjklsd
 tamirtest fkljsdkfjksd
 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3 kjerjferfjk
 jfhk3jhfjh3diojsffffffffffffffffffffffffffffffffffffffffffr83489ru834
 pofidofii349r34utfujre
 flodsoksopifiowesmash> smash>  tamirtest lkjfekjlfjk
 tamirtest fkljsdkfjksd
smash> smash> tamirtest 1234                                                                               
tamirtest 1234
tamirtest 1234
tamirtest 5678                                                                               
tamirtest 5678                                                                               
tamirtest 5678                                                                               
tamirtest 1234
tamirtest 5678                                                                               
tamirtest 5678                                                                               
tamirtest 1234
smash> tamirtest 1234                                                                               
tamirtest 1234
tamirtest 1234
tamirtest 5678                                                                               
tamirtest 5678                                                                               
tamirtest 5678                                                                               
tamirtest 1234
tamirtest 5678                                                                               
tamirtest 5678                                                                               
tamirtest 1234smash> smash> smash> smash> [1] timeout 110 sleep 20 & : 7292 14 secs
[2] sleep 150 : 7293 7 secs (stopped)
[3] timeout 15 sleep 5& : 7306 0 secs
[4] timeout 10 sleep 16& : 7308 0 secs
[5] timeout 30 sleep 7 & : 7309 0 secs
smash> smash: got an alarm
smash: got an alarm
smash: timeout 10 sleep 16& timed out!
smash: got an alarm
smash: got ctrl-C
smash: got ctrl-Z
smash pid is 7291
smash> smash pid is 7291
smash> smash: got ctrl-Z
[2] sleep 150 : 7293 23 secs (stopped)
smash> smash: got an alarm
sleep 150 : 7293
smash: got ctrl-Z
smash: process 7293 was stopped
smash> [2] sleep 150 : 7293 0 secs (stopped)
smash> smash> smash> sleep 15& : 7327
smash: got ctrl-Z
smash: process 7327 was stopped
smash> [2] sleep 150 : 7293 5 secs (stopped)
[3] sleep 15& : 7327 5 secs (stopped)
[4] sleep 100& : 7328 5 secs
smash> sleep 15& : 7327
smash> [2] sleep 150 : 7293 11 secs (stopped)
[3] sleep 15& : 7327 11 secs
[4] sleep 100& : 7328 11 secs
smash> smash> signal number 9 was sent to pid 7328
smash> [2] sleep 150 : 7293 21 secs (stopped)
[5] timeout 20 sleep 150& : 7334 10 secs
smash> smash error: kill: job-id 3 does not exist
smash> smash> smash error: kill: job-id 3 does not exist
smash> smash: got an alarm
smash: timeout 20 sleep 150& timed out!
[2] sleep 150 : 7293 34 secs (stopped)
[6] sleep 200& : 7341 8 secs
smash> smash> smash> smash> [2] sleep 150 : 7293 34 secs (stopped)
[6] sleep 200& : 7341 8 secs
[7] sleep 100& : 7347 0 secs
[8] sleep 200& : 7348 0 secs
[9] sleep 500 & : 7349 0 secs
smash> signal number 19 was sent to pid 7348
smash> smash> signal number 18 was sent to pid 7348
smash> [2] sleep 150 : 7293 47 secs (stopped)
[6] sleep 200& : 7341 21 secs
[7] sleep 100& : 7347 13 secs
[8] sleep 200& : 7348 13 secs
[9] sleep 500 & : 7349 13 secs
[10] sleep 200& : 7354 8 secs
smash> sleep 200& : 7354
smash: got ctrl-Z
smash: process 7354 was stopped
smash> sleep 100& : 7347
smash: got an alarm
smash: got ctrl-Z
smash: process 7347 was stopped
smash> sleep 200& : 7348
smash: got ctrl-C
smash: process 7348 was killed
smash> [2] sleep 150 : 7293 69 secs (stopped)
[6] sleep 200& : 7341 43 secs
[7] sleep 100& : 7347 8 secs (stopped)
[9] sleep 500 & : 7349 35 secs
[10] sleep 200& : 7354 15 secs (stopped)
smash> smash> /tmp/smash_test
smash> smash error: cd: OLDPWD not set
smash> smash error: cd: OLDPWD not set
smash> /tmp/smash_test
smash> smash> smash: got an alarm
smash: timeout 10 sleep 150& timed out!
did I pass the test?
smash> smash> did I pass the test?
I hope so
smash> Bye Bye
smash> smash: sending SIGKILL signal to 5 jobs:
7293: sleep 150
7341: sleep 200&
7347: sleep 100&
7349: sleep 500 &
7354: sleep 200&
